packages/start-slicemachine/src/lib/ensureFSAssetsState.ts

```diff
@@ -0,0 +1,173 @@
+import * as path from "node:path";
+import * as fsSync from "node:fs";
+import { SliceMachineManager } from "@slicemachine/manager";
+import {
+	SharedSliceContent,
+	Document,
+} from "@prismicio/types-internal/lib/content";
+import { DocumentMock, SharedSliceMock } from "@prismicio/mocks";
+
+const MOCKS_FILE_NAME = "mocks.json";
+const MOCK_CONFIG_FILE_NAME = "mock-config.json";
+
+const createPathToDeprecatedLibrary = (cwd: string) =>
+	path.join(cwd, ".slicemachine");
+const createPathToCustomTypesAssets = (cwd: string) =>
+	path.join(createPathToDeprecatedLibrary(cwd), "assets", "customtypes");
+
+const safeUnlink = (pathToUnlink: string, type: "file" | "folder") => {
+	try {
+		if (type === "file") {
+			fsSync.unlinkSync(pathToUnlink);
+		} else {
+			fsSync.rmSync(pathToUnlink, { recursive: true });
+		}
+	} catch (_) {}
+};
+
+const ensureOrGenerateSliceScreenshot = (
+	variationsIDs: string[],
+	targetPathToSliceFolder: string,
+	deprecatedPathToSliceAssets: string,
+) => {
+	variationsIDs.forEach((variationID) => {
+		const targetPathToVariationScreenshot = path.join(
+			targetPathToSliceFolder,
+			`screenshot-${variationID}.png`,
+		);
+		if (!fsSync.existsSync(targetPathToVariationScreenshot)) {
+			const deprecatedPathToVariationScreenshot = path.join(
+				deprecatedPathToSliceAssets,
+				variationID,
+				"preview.png",
+			);
+			if (fsSync.existsSync(deprecatedPathToVariationScreenshot)) {
+				fsSync.renameSync(
+					deprecatedPathToVariationScreenshot,
+					targetPathToVariationScreenshot,
+				);
+			}
+		}
+	});
+};
+
+const ensureOrGenerateMockFile = (
+	targetPathToMocks: string,
+	deprecatedPathToMocks: string,
+	validator: (str: string) => boolean,
+	generate: () => Record<string, unknown>,
+) => {
+	try {
+		const rawMockContent = (() => {
+			if (fsSync.existsSync(targetPathToMocks)) {
+				return fsSync.readFileSync(targetPathToMocks, "utf-8");
+			}
+			if (fsSync.existsSync(deprecatedPathToMocks)) {
+				return fsSync.readFileSync(deprecatedPathToMocks, "utf-8");
+			}
+
+			return null;
+		})();
+
+		const regeneratedMocksString = JSON.stringify(generate(), null, 2);
+		const isValidMock = (() => {
+			try {
+				return rawMockContent !== null && validator(rawMockContent);
+			} catch {
+				return false;
+			}
+		})();
+
+		if (!isValidMock) {
+			fsSync.writeFileSync(targetPathToMocks, regeneratedMocksString);
+		} else if (rawMockContent) {
+			fsSync.writeFileSync(targetPathToMocks, rawMockContent);
+		} else {
+			fsSync.writeFileSync(targetPathToMocks, regeneratedMocksString);
+		}
+	} catch (error) {}
+};
+
+export const ensureFSAssetsState = async (
+	manager: SliceMachineManager,
+): Promise<void> => {
+	try {
+		const allSlices = await manager.slices.readAllSlices();
+		const sharedSlices = allSlices.models.reduce(
+			(o, slice) => ({ ...o, [slice.model.id]: slice.model }),
+			{},
+		);
+
+		allSlices.models.forEach((c) => {
+			const targetPathToSliceFolder = path.join(
+				manager.cwd,
```

Comment:
Don't we have helpers for this?

Reply:
Not really because it's plugins that are handling that now, and this is a special case were we migrate old data structure. I think it's simpler kept that way